flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/sink/global/AbstractCommitterOperator.java

```diff
@@ -0,0 +1,155 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.connector.sink.global;
+
+import org.apache.flink.api.common.state.ListState;
+import org.apache.flink.api.common.state.ListStateDescriptor;
+import org.apache.flink.api.common.typeutils.base.array.BytePrimitiveArraySerializer;
+import org.apache.flink.api.connector.sink2.Sink;
+import org.apache.flink.core.io.SimpleVersionedSerializer;
+import org.apache.flink.runtime.state.StateInitializationContext;
+import org.apache.flink.runtime.state.StateSnapshotContext;
+import org.apache.flink.streaming.api.connector.sink2.CommittableMessage;
+import org.apache.flink.streaming.api.connector.sink2.CommittableWithLineage;
+import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
+import org.apache.flink.streaming.api.operators.BoundedOneInput;
+import org.apache.flink.streaming.api.operators.ChainingStrategy;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.api.operators.util.SimpleVersionedListState;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.util.function.SerializableSupplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+
+/** An operator that processes committables of a {@link Sink}. */
+public abstract class AbstractCommitterOperator<IN, CommT>
+        extends AbstractStreamOperator<CommittableMessage<IN>>
+        implements OneInputStreamOperator<CommittableMessage<IN>, CommittableMessage<IN>>,
+                BoundedOneInput {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(GlobalCommitterOperator.class);
+
+    /** Record all the inputs until commit. */
+    private final Deque<IN> inputs = new ArrayDeque<>();
+
+    /** The operator's state descriptor. */
+    private static final ListStateDescriptor<byte[]> STREAMING_COMMITTER_RAW_STATES_DESC =
+            new ListStateDescriptor<>(
+                    "streaming_committer_raw_states", BytePrimitiveArraySerializer.INSTANCE);
+
+    /** Group the committable by the checkpoint id. */
+    private final NavigableMap<Long, List<CommT>> committablesPerCheckpoint;
+
+    /** The committable's serializer. */
+    private final SerializableSupplier<SimpleVersionedSerializer<CommT>> committableSerializer;
```

Comment:
Just curious, why not use `CommittableSerializer`?

Reply:
One is `CommittableSerializer`, one is global `ManifestCommittableSerializer`.
And this should be abstract class.